[new-product] Salt (#3358)
captn3m0/endoflife.date: products/salt.md

---

title: Salt
category: server-app
tags: python-runtime
iconSlug: saltproject
permalink: /salt
alternate_urls:
-   /saltstack
versionCommand: salt --version
releasePolicyLink: https://docs.saltproject.io/salt/install-guide/en/latest/topics/salt-version-support-lifecycle.html
releaseImage: https://docs.saltproject.io/salt/install-guide/en/latest/_images/salt-release-timeline.png
changelogTemplate: https://docs.saltproject.io/en/latest/topics/releases/__LATEST__.html#changelog
eolColumn: CVE & Critical Support
activeSupportColumn: true
releaseColumn: true
releaseDateColumn: true

auto:
# https://rubular.com/r/ELfj6SIxS0dZk7
-   git: https://github.com/saltstack/salt.git
    regex: ^v(?<version>([1-9]\d*)(\.\d+){0,3})$
    template: "{{version}}"

identifiers:
  - repology: salt
  - purl: pkg:oci/docker-salt-master?repository_url=ghcr.io/cdalvaro
  - purl: pkg:docker/saltstack/salt

releases:

-   releaseCycle: "3006"
    releaseDate: 2023-04-18
    support: 2024-04-18
    eol: 2025-04-18
    latest: "3006.1"
    latestReleaseDate: 2023-05-05
    lts: true

>     link: https://docs.saltproject.io/en/latest/topics/releases/3006.0.html#changelog

Reviewer: ```suggestion
```

The `changelogTemplate` should be enough.

-   releaseCycle: "3005"
    releaseDate: 2022-08-25
    support: 2023-08-25
    eol: 2024-02-25
    latest: "3005.1"
    latestReleaseDate: 2022-11-08

>     link: https://docs.saltproject.io/en/latest/topics/releases/3005.1.html

Reviewer: ```suggestion
```

The `changelogTemplate` should be enough.

-   releaseCycle: "3004"
    releaseDate: 2021-10-18
    support: 2022-10-18
    eol: 2023-04-18
    latest: "3004.2"
    latestReleaseDate: 2022-06-21

>     link: https://docs.saltproject.io/en/latest/topics/releases/3004.html

Reviewer: ```suggestion
```

The `changelogTemplate` should be enough.

---

> [Salt](https://saltproject.io/index.html) is software to automate the management and configuration of any infrastructure or application at scale.

> ## Support Lifecycle

Reviewer: ```suggestion
```

A title is not needed as there is only one section in this description.

Beginning with the 3006 release of Salt, the Salt Project will follow an LTS/STS release strategy. Under this strategy, the Salt Project will release one LTS (long-term support) release of Salt per year and one STS (short-term support) release each year.

> LTS prioritize stability, and are supported for 2 years. STS releases provide access to newer features in between LTS releases, and are only supported for 6 months.

Reviewer: ```suggestion
Beginning with the 3006 release of Salt, the Salt Project is following an LTS/STS release strategy. Under this strategy, there is one long-term support (LTS) and one short-term support (STS) release a year.

LTS prioritize stability. They are released usually in the first 4 months of the year and receive one year of full support, with bug/security fixes, followed by one year of security support.

STS releases provide access to newer features in between LTS. They are released around the midpoint the year and receive three months of full support, with new OS support and bug/security fixes, followed by three months of security support.
```

Added a little bit more details. WDYT ?

Author: Looks good to me!

---

title: Salt
category: server-app

> tags: salt

Reviewer: ```suggestion
tags: python-runtime
```

We're using a limited set of tags a favor reusable ones, cf. tags on https://endoflife.date/ansible.

iconSlug: saltproject

> permalink: /salt

Reviewer: ```suggestion
permalink: /salt
alternate_urls:
-   /saltstack
```

According to https://en.wikipedia.org/wiki/Salt_(software) :

> Salt (sometimes referred to as SaltStack) is a Python-based, [open-source software](https://en.wikipedia.org/wiki/Open-source_software) for event-driven IT automation, remote task execution, and [configuration management](https://en.wikipedia.org/wiki/Configuration_management).

versionCommand: salt --version
releasePolicyLink: https://docs.saltproject.io/salt/install-guide/en/latest/topics/salt-version-support-lifecycle.html
releaseImage: https://docs.saltproject.io/salt/install-guide/en/latest/_images/salt-release-timeline.png
changelogTemplate: https://docs.saltproject.io/en/latest/topics/releases/__LATEST__.html#changelog
eolColumn: CVE & Critical Support

> activeSupportColumn: Active Support

Reviewer: ```suggestion
activeSupportColumn: true
```

`Active Support` is the default label.

releaseColumn: true
releaseDateColumn: true

auto:

>   - git: https://github.com/saltstack/salt.git

Reviewer 2: This catches all the stable releases correctly:

```suggestion
# https://rubular.com/r/ELfj6SIxS0dZk7
-   git: https://github.com/saltstack/salt.git
    regex: ^v(?<version>([1-9]\d*)(\.\d+){0,3})$
    template: "{{version}}"
```

identifiers:

>   - repology: salt

Reviewer 2: Add a few popular docker images, including the official one:

```suggestion
  - repology: salt
  - purl: pkg:oci/docker-salt-master?repository_url=ghcr.io/cdalvaro
  - purl: pkg:docker/saltstack/salt
```

    eol: 2024-02-25
    latest: "3005.1"
    latestReleaseDate: 2022-11-08
    link: https://docs.saltproject.io/en/latest/topics/releases/3005.1.html

-   releaseCycle: "3004"
    releaseDate: 2021-10-18
    support: 2022-10-18
    eol: 2023-04-18
    latest: "3004.2"
    latestReleaseDate: 2022-06-21
    link: https://docs.saltproject.io/en/latest/topics/releases/3004.html

---

> [Salt](https://saltproject.io/index.html) is software to automate the management and configuration of any infrastructure or application at scale.

## Support Lifecycle
Beginning with the 3006 release of Salt, the Salt Project will follow an LTS/STS release strategy. Under this strategy, the Salt Project will release one LTS (long-term support) release of Salt per year and one STS (short-term support) release each year. The LTS release will receive bug fix releases for a longer period of time than STS releases.

The purpose of the LTS release is to provide users with a stable version of Salt for a longer period (2 years). The purpose of the STS release is to provide a feature release to users who need or prefer access to the latest features in between LTS releases.

## Support Comparison

> |                      |                                                                                                                                LTS release                                                                                                                               |                                                                                                   STS release                                                                                                   |

Reviewer 2: I find tables hard to maintain, is it okay if I were to split this into separate sections instead (Support, Release Cadence)

Author: I think we can drop the table all together, it is documented on the salt website.
